logging.go

```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"io"
+	"time"
+)
+
+type clock interface {
+	now() time.Time
+}
+
+type realTimeClock struct{}
+
+func (c *realTimeClock) now() time.Time {
+	return time.Now()
+}
+
+type rfcPrefixWriter struct {
```

Comment:
Would it be possible to use an interface here? Maybe with some kind of factory? That would make it possible to add different formats later

(for example, I'm thinking of a logger which uses JSON format for easier parsing).

Reply:
When we choose to implement structured logging, we should consider using dedicated libraries like [Zerolog](https://github.com/rs/zerolog).

Reply:
Would it make sense to do it correctly now?
(And plan a major release for it?)